openedx/core/djangoapps/catalog/utils.py

```diff
@@ -243,9 +243,22 @@ def get_course_runs_for_course(course_uuid):
         return []
 
 
-def get_visible_course_runs_for_entitlement(entitlement):
+def get_mock_session_for_entitlement(entitlement):
     """
-    Returns only the course runs that the user can currently enroll in.
+    This function is used to pass pseudo-data to the front end, returning a single session, regardless of whether that
+    session is currently available.
+
+    First tries to return the first available session, followed by the first session regardless of availability.
+    Returns None if there are no sessions for that course.
+    """
+    sessions_for_course = get_course_runs_for_course(entitlement.course_uuid)
+    available_sessions = get_fulfillable_course_runs_for_entitlement(entitlement, sessions_for_course)
+    return available_sessions[0] if available_sessions else (sessions_for_course[0] if sessions_for_course else None)
```

Comment:
Why put this all in one line?  It would be more clear broken down into a few lines.

```
available_sessions = get_fulfillable_course_runs_for_entitlement(entitlement, sessions_for_course)
if available_sessions and len(available_sessions) > 0:
  return available_sessions[0]

sessions_for_course = get_course_runs_for_course(entitlement.course_uuid)
if sessions_for_course and len(sessions_for_course) > 0:
  return sessions_for_course[0]

return None
```

This will also remove the call to the method and potentially DB for get_course_runs_for_course

Reply:
The available_sessions variable needs to sessions_for_course variable to render, so there wouldn't be any time speedup, but I can clean up for code clarity.